purple/key.py

```diff
@@ -20,3 +20,25 @@ def just_released(self):
     def update(self):
         self._prior = self.current
         self.current = not self._pin.value
+
+class MatrixKey:
+    def __init__(self, diode_pin, wire_pin):
+        self._prior = False
+        self.current = False
+        self._diode_pin = diode_pin
+        self._wire_pin = wire_pin
+    
+    @property
+    def just_pressed(self):
+        return self.current and not self._prior
+    
+    @property
+    def just_released(self):
+        return self._prior and not self.current
+
+    def update(self):
+        self._prior = self.current
+        self._wire_pin.value = True
```

Comment:
Did you intend to set `self._wire_pin.value` to True and then right back to False? Does this somehow help `self.current = self._diode_pin.value`? It's hard to tell without seeing your actual usage.

Reply:
With the matrix wiring, you set, depending on the orientation of the diode, the row or column high and check if the voltage is high on the other to see if the key is depressed.  There might be better naming of these to indicate their function, e.g. _signal_pin and _measure_pin.